### ISSUE TITLE
Incorrect behavior on items change

### ISSUE THREAD

**Issue by fedorl** [rintoj/angular2-virtual-scroll#40]
May be I miss something, I'm trying to use with angular 4 ngfor and updating items with mobx autorun.

Initially: one item (with zero doesn't work because of [this](https://github.com/rintoj/angular2-virtual-scroll/issues/39)) is displayed correctly.

Setting [items] to another array instance (initial concatenated with 15 items), all contained in viewport: not all items are displayed and shows vertical scroll bar that shouldn't be there. 

It seems that countItemsPerRow is called only once after replace and at the time of the call there this.items.length is 16 but this.contentElementRef.nativeElement.children.length is 1. So it uses 1 item per row in calculations resulting in incorrect values.

**Comment by fedorl** [comment]
Meantime the quick fix below seems working when childWidth is present, but requires more investigation on update life cycle.
`
  private countItemsPerRow() {
    if (this.childWidth) {
      const el = this.element.nativeElement;
      return (el.clientWidth - this.scrollbarWidth) / this.childWidth;
    }
   ....
`

**Comment by rintoj** [comment]
@fedorl, can you check again I have fixed #39.

**Comment by fedorl** [comment]
@rintoj Thanks for fixing! I'll check next week, have a tight schedule now. 

Does it fix #40 as well or #39 only?

**Comment by rintoj** [comment]
The fix should work for both #40 and #39. But please check and if the problem still exists, please reopen this issue. Thanks.

**Comment by fedorl** [comment]
The fix seems to solve #40 and #39 (+ another issue of wrong item indexes).
